doc/fq.1

```diff
@@ -0,0 +1,392 @@
+.TH FQ "1" "July 2023"
+
+.SH NAME
+fq \- jq for binary formats
+
+.SH SYNOPSIS
+\fBfq\fR [\fI\,OPTION\/\fR]... [--] \fI\,EXPR [\fI\,FILE\/\fR]...
+
+.SH DESCRIPTION
+\fBfq\fR is inspired by the well known jq tool and language that allows you to work with binary formats the same way you would using jq. In addition it can present data like a hex viewer, transform, slice and concatenate binary data. It also supports nested formats and has an interactive REPL with auto-completion.
+.P
+It was originally designed to query, inspect and debug media codecs and containers like mp4, flac, mp3, jpeg. But since then it has been extended to support a variety of formats like executables, packet captures (with TCP reassembly) and serialization formats like JSON, YAML, XML, ASN1 BER, Avro, CBOR, protobuf. In addition it also has functions to work with URLs, convert to/from hex, number bases, search for things etc.
+.P
+In summary it aims to be jq, hexdump, dd and gdb for files combined into one.
+
+.SH INVOKING FQ
+.SS Command Line Arguments
+Most of jq's CLI arguments work with fq. But there are some additional ones specific to fq:
+.TP
+\fB-d\fR, \fB--decode\fR \fI\,FORMAT\/\fR
+Use \fI\,FORMAT\/\fR to decode instead of trying to auto-detect format.
+.TP
+\fB-i\fR, \fB--repl\fR [\fI\,INPUT\/\fR]...
+Use interactive REPL.
+
+Can be used with no input, one and multiple inputs, for example just \fBfq -i\fR starts a REPL with null input, \fBfq -i 123\fR with the number 123 as input, \fBfq -i . a b\fR with two files as input. This also works with \fB--slurp\fR. In the REPL it is also possible to start a sub-REPL(s) by ending a query with \fB<query> | repl\fR, use ctrl-D to exit the sub-REPL. The sub-REPL will evaluate separately on each output from the query it was started. Use \fB[<query>] | repl\fR if you want to "slurp" into an array.
+.TP
+\fB-o\fR, \fB--options\fR <\fIKEY\fR=\fIVALUE\fR|@\fIPATH\fR>
+\fIKEY\fR is name of option
+
+\fIVALUE\fR will be interpreted as a JSON value if possible otherwise a string, for example \fB-o name=abc\fR and \fB-o name='"abc"'\fR are equivalent.
+
+@\fBPATH\fR will read string from file at \fBPATH\fR.
+
+Specifying a global option or a format option, for example, \fB-o decode_samples=false\fR would, for some container decoders like mp4 and matroska, disable decoding of samples.
+.TP
+\fB-V\fR, \fB--value-output\fR
+Output JSON value instead of decode tree. Use \fB-Vr\fR if you want raw string (no quotes).
+.TP
+\fB-M\fR
+Disable coloured output.
+.TP
+\fB-o <NAME>=<VALUE>\fR
+Set a decoding/encoding option. A list of supported options are -
+.RS
+.P
+\fBbits_format\fR - string, md5, hex, base64, truncate, snippet
+.RS
+Controls the representation of raw binary in the output JSON
+.RE
+.RE
+.SS Supported Formats
+Use \fBfq --help formats\fR to get a list of supported formats
```

Comment:
If we're nog going to auto generate this man page maybe this could instead give example how to list supported formwats?

Reply:
I'm not sure I understand this question, isn't `fq --help formats` the way to list supported formats?

Reply:
Ah sorry i totally missed that, only saw the list of formats :) and what i was wondering was if we should include the list in the man page or not as it will have to be manually kept up to date if we don't update/generate it with `make doc` etc.